tests/AstrolabeWorkloadExecutor/Program.cs

```diff
@@ -0,0 +1,114 @@
+﻿/* Copyright 2020-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.IO;
+using System.Threading;
+using MongoDB.Bson;
+
+namespace WorkloadExecutor
+{
+    class Program
+    {
+        static long __numberOfSuccessfulOperations;
+        static long __numberOfFailedOperations;
+        static long __numberOfOperationErrors;
+
+        static void Main(string[] args)
+        {
+            var connectionString = args[0];
+            var driverWorkload = BsonDocument.Parse(args[1]);
+
+            var cancellationTokenSource = new CancellationTokenSource();
+            ConsoleCancelEventHandler cancelHandler = (o, e) => HandleCancel(e, cancellationTokenSource);
+
+            var resultsDir = Environment.GetEnvironmentVariable("RESULTS_DIR");
+            var resultsPath = resultsDir == null ? "results.json" : Path.Combine(resultsDir, "results.json");
+            Console.WriteLine($"dotnet main> Results will be written to {resultsPath}..." );
+
+            try
+            {
+                Console.CancelKeyPress += cancelHandler;
+
+                Console.WriteLine("dotnet main> Starting workload executor...");
+
+                if (!bool.TryParse(Environment.GetEnvironmentVariable("ASYNC"), out bool async))
+                {
+                    async = true;
+                }
+
+                ExecuteWorkload(connectionString, driverWorkload, async, cancellationTokenSource.Token);
+            }
+            finally
+            {
+                Console.CancelKeyPress -= cancelHandler;
+                Console.WriteLine("dotnet main finally> Writing final results file");
+                var resultsJson = ConvertResultsToJson();
+                Console.WriteLine(resultsJson);
+#if NETCOREAPP2_1
+                File.WriteAllTextAsync(resultsPath, resultsJson).Wait();
+#else
+                File.WriteAllText(resultsPath, resultsJson);
+#endif
+            }
+        }
+
+        static string ConvertResultsToJson()
+        {
+                var resultsJson =
+                @"{ " +
+                $"  \"numErrors\" : {Interlocked.Read(ref __numberOfOperationErrors)}, " +
+                $"  \"numFailures\" : {Interlocked.Read(ref __numberOfFailedOperations)}, " +
+                $"  \"numSuccesses\" : {Interlocked.Read(ref __numberOfSuccessfulOperations)}  " +
```

Comment:
Why `Interlocked`? I don't believe there is any multi-threading going on in this program.